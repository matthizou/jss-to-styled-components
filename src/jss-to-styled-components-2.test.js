const pluginTester = require('babel-plugin-tester').default
const jssToStyledComponentsPlugin = require('./babel-plugin-jss-to-styled-components')
const { processCss } = require('./processCss')

const findCss = processCss({ filePath: './assets/360-mini.css' }).findCss

// return
pluginTester({
  pluginName: 'jssToStyledComponentsPlugin',
  plugin: jssToStyledComponentsPlugin,
  babelOptions: require('../babel.config.js'),
  snapshot: true,
  tests: [
    // Use of template literal syntax in jsx classes
    {
      code: `
      import injectSheet from 'react-jss'
      const color = '#EEE'
      const styles = {
        container: {
            marginTop: "12px",
            borderTop: \`1px solid $\{color}\`,
            marginBottom: "-24px",
          },
      }

      const HeaderFooterCard = ({ classes }) => (
        <section className={classes.container}>🥝 Object values</section>
      )
    `,
      pluginOptions: {
        findCss,
      },
    },
  ],
})
