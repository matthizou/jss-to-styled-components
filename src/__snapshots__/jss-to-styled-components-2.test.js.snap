// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jssToStyledComponentsPlugin 1. jssToStyledComponentsPlugin: 1. jssToStyledComponentsPlugin 1`] = `

import injectSheet from 'react-jss'
const color = '#EEE'
const styles = {
  container: {
      marginTop: "12px",
      borderTop: \`1px solid \${color}\`,
      marginBottom: "-24px",
    },
}

const HeaderFooterCard = ({ classes }) => (
  <section className={classes.container}>🥝 Object values</section>
)

      ↓ ↓ ↓ ↓ ↓ ↓

import styled from 'styled-components'
const color = '#EEE'
const Container = styled.section\`
  margin-top: 12px;
  border-top: 1px solid \${color};
  margin-bottom: -24px;
\`

const HeaderFooterCard = ({ classes }) => (
  <Container>🥝 Object values</Container>
)


`;
